File: frontend/src/types/threat_intel.ts
export interface CISAKEVEntry {
  id: number
  cve_id: string
  vendor_project: string
  product: string
  vulnerability_name: string
  date_added: string
  short_description: string
  required_action: string
  due_date: string
  notes: string
}

export interface EPSSScore {
  cve_id: string
  score: number
  percentile: number
  date: string
}

export interface ThreatFeedSync {
  id: number
  feed_name: string
  last_sync: string
  item_count: number
  new_items: number
}

export interface ThreatFeedsStatus {
  feodo?: ThreatFeedSync
  urlhaus?: ThreatFeedSync
}

export interface ThreatFeedSyncResult {
  new_iocs: number
  skipped_duplicates: number
  total_in_feed?: number
  total_online?: number
}

export interface ImportResult {
  imported: number
  skipped_duplicates: number
  skipped_invalid: number
  errors: string[]
}

File: backend/internal/models/threat_intel.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type CISAKEVEntry struct {
	gorm.Model
	CveID             string `gorm:"uniqueIndex" json:"cve_id"`
	VendorProject     string `json:"vendor_project"`
	Product           string `json:"product"`
	VulnerabilityName string `json:"vulnerability_name"`
	DateAdded         string `json:"date_added"`
	ShortDescription  string `json:"short_description"`
	RequiredAction    string `json:"required_action"`
	DueDate           string `json:"due_date"`
	Notes             string `json:"notes"`
}

type EPSSScore struct {
	gorm.Model
	CveID      string    `gorm:"uniqueIndex" json:"cve_id"`
	Score      float64   `json:"score"`
	Percentile float64   `json:"percentile"`
	Date       string    `json:"date"`
	FetchedAt  time.Time `json:"fetched_at"`
	ExpiresAt  time.Time `json:"expires_at"`
}

type ThreatFeedSync struct {
	gorm.Model
	FeedName  string    `gorm:"uniqueIndex" json:"feed_name"`
	LastSync  time.Time `json:"last_sync"`
	ItemCount int       `json:"item_count"`
	NewItems  int       `json:"new_items"`
}
